src/pokemon.c

```diff
@@ -3776,12 +3782,21 @@ u8 GetMonsStateToDoubles(void)
     return (aliveCount > 1) ? PLAYER_HAS_TWO_USABLE_MONS : PLAYER_HAS_ONE_USABLE_MON;
 }
 
-u8 GetAbilityBySpecies(u16 species, bool8 abilityNum)
+u8 GetAbilityBySpecies(u16 species, u8 abilityNum, u8 lockedAbility)
 {
-    if (abilityNum)
-        gLastUsedAbility = gSpeciesInfo[species].abilities[1];
-    else
-        gLastUsedAbility = gSpeciesInfo[species].abilities[0];
+    switch (abilityNum)
+    {
+    case 0:
+    case 1:
+        gLastUsedAbility = gSpeciesInfo[species].abilities[abilityNum];
+        break;
+    case 2: //hidden ability
```

Comment:
Should we set `gLastUsedAbility` to a default value in case 2 and at the end to avoid potential unexpected behavior?

Reply:
I wonder why ghoulslash made it this way. I agree and slightly changed it.